tests/fixtures/saas/sparkpost_fixtures.py

```diff
@@ -0,0 +1,121 @@
+import uuid
+from typing import Any, Dict, Generator
+
+import pydash
+import pytest
+import requests
+
+from tests.ops.integration_tests.saas.connector_runner import (
+    ConnectorRunner,
+    generate_random_email,
+)
+from tests.ops.test_helpers.saas_test_utils import poll_for_existence
+from tests.ops.test_helpers.vault_client import get_secrets
+
+secrets = get_secrets("sparkpost")
+
+
+@pytest.fixture(scope="session")
+def sparkpost_secrets(saas_config) -> Dict[str, Any]:
+    return {
+        "domain": pydash.get(saas_config, "sparkpost.domain") or secrets["domain"],
+        "api_key": pydash.get(saas_config, "sparkpost.api_key") or secrets["api_key"],
+    }
+
+
+@pytest.fixture(scope="session")
+def sparkpost_identity_email(saas_config) -> str:
+    return (
+        pydash.get(saas_config, "sparkpost.identity_email") or secrets["identity_email"]
+    )
+
+
+@pytest.fixture(scope="function")
+def sparkpost_erasure_recipient_id():
+    return f"{uuid.uuid4().hex}"
+
+
+@pytest.fixture
+def sparkpost_erasure_identity_email() -> str:
+    return generate_random_email()
+
+
+class SparkPostClient:
+    headers: object = {}
+    base_url: str = ""
+
+    def __init__(self, secrets: Dict[str, Any]):
+        self.base_url = f"https://{secrets['domain']}"
+        self.headers = {
+            "Authorization": f"Basic {secrets['api_key']}",
+        }
+
+    def get_recipient(self, recipient_id: str, email: str) -> requests.Response:
+        body = {
+            "filterGroups": [
+                {
+                    "filters": [
+                        {
+                            "value": email,
+                            "propertyName": "address.email",
+                            "operator": "EQ",
+                        }
+                    ]
+                }
+            ]
+        }
+        recipient_response: requests.Response = requests.post(
+            url=f"{self.base_url}/api/v1/recipient-lists/{recipient_id}?show_recipients=true",
+            json=body,
+            headers=self.headers,
+        )
+        assert recipient_response.ok
+        return recipient_response.json()
+
+    def create_recipient(self, email, uuid):
```

Comment:
Can you update this to add a recipient to an existing recipient list instead of creating a new recipient list for every test? Let's avoid creating data that won't be deleted by the erasure test.